index.js

```diff
@@ -10,10 +10,17 @@
 var typeOf = require('kind-of');
 
 module.exports = function isNumber(num) {
+  if (!num && num !== 0) return false;
```

Comment:
Is this line necessary? (I'm really asking, I might be missing something) seems like we would catch that with existing conditions